src/loss_functions/losses.py

```diff
@@ -36,15 +36,15 @@ def forward(self, x, y):
 
         # Asymmetric Focusing
         if self.gamma_neg > 0 or self.gamma_pos > 0:
-            if self.disable_torch_grad_focal_loss:
-                torch._C.set_grad_enabled(False)
+            # if self.disable_torch_grad_focal_loss:
+            #     torch._C.set_grad_enabled(False)
             pt0 = xs_pos * y
             pt1 = xs_neg * (1 - y)  # pt = p if t > 0 else 1-p
             pt = pt0 + pt1
             one_sided_gamma = self.gamma_pos * y + self.gamma_neg * (1 - y)
             one_sided_w = torch.pow(1 - pt, one_sided_gamma)
-            if self.disable_torch_grad_focal_loss:
-                torch._C.set_grad_enabled(True)
+            # if self.disable_torch_grad_focal_loss:
+            #     torch._C.set_grad_enabled(True)
```

Comment:
**Suggestion:** It seems like the code lines disabling and enabling the gradient computation in PyTorch have been commented out. If you want to conditionally disable gradient computations, consider using the `torch.no_grad()` context manager instead of directly manipulating the `_C` module. This would make the code more readable and less prone to errors. [best practice]
```suggestion
            if self.disable_torch_grad_focal_loss:
                with torch.no_grad():
                    pt0 = xs_pos * y
                    pt1 = xs_neg * (1 - y)  # pt = p if t > 0 else 1-p
                    pt = pt0 + pt1
                    one_sided_gamma = self.gamma_pos * y + self.gamma_neg * (1 - y)
                    one_sided_w = torch.pow(1 - pt, one_sided_gamma)
            else:
                pt0 = xs_pos * y
                pt1 = xs_neg * (1 - y)  # pt = p if t > 0 else 1-p
                pt = pt0 + pt1
                one_sided_gamma = self.gamma_pos * y + self.gamma_neg * (1 - y)
                one_sided_w = torch.pow(1 - pt, one_sided_gamma)
            loss *= one_sided_w
```